### PR TITLE
Atizaz abid

### PR DESCRIPTION
**Pull request by Atizaz-Abid**
Added the models and dash board and get categories api

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+
+
+
+class TimeStampedModel(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Always use small letters with `_`. Variable name must be `is_remove`

File: models.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+class Author(models.Model):
+    user = models.ForeignKey(User, related_name='author',db_index=True)
+    avatar = models.ImageField(upload_to='gallery/avatar/%Y/%m/%d',
+                               null=True,
+                               blank=True,
+                               help_text="Upload your photo for Avatar")
+    about = models.TextField()
+    website = models.URLField(max_length=200, blank=True, null=True)
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Same as above

File: models.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    def __str__(self):
+        return self.user.username
+
+    def get_absolute_url(self):
+        return reverse('author_posts_page',
+                       kwargs={'username': self.user.username})
+
+    class Meta:
+        verbose_name = 'Detail Author'
+        verbose_name_plural = 'Authors'
```
Comment (tahirawan4) [inline models.py]: We don't need these meta variables, as author plural is authors by default. Remove it please

File: models.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    isRemove = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def get_total_posts(self):
+        return Feeds.objects.filter(tags__pk=self.pk).count()
+
+    class Meta:
```
Comment (tahirawan4) [inline models.py]: We won't need here too.

File: models.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        verbose_name = 'Detail Tag'
+        verbose_name_plural = 'Tags'
+
+
+class FeedsApiUrl(models.QuerySet):
+    api_url = models.TextField(null=True)
+    description = models.TextField(null=True)
+    time = models.ForeignKey(TimeStampedModel,on_delete=models.CASCADE,related_name='time_stamp',db_index=True)
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Update this one.

File: models.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+                              help_text='Optional cover post')
+    description = RedactorField()
+    tags = models.ManyToManyField('Tag')
+    keywords = models.CharField(max_length=200, null=True, blank=True,
+                                help_text='Keywords sparate by comma.')
+    meta_description = models.TextField(null=True, blank=True)
+
+    publish = models.BooleanField(default=True)
+    objects = FeedsApiUrl.as_manager()
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Update it

File: models.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        ordering = ["-created"]
+
+
+class Page(TimeStampedModel):
+    author = models.ForeignKey(Author, related_name='author_page')
+    title = models.CharField(max_length=200)
+    slug = models.SlugField(max_length=200, unique=True)
+    description = RedactorField()
+    publish = models.BooleanField(default=True)
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Update it

File: models.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    def __str__(self):
+        return self.title
+
+    # this will be an error in /admin
+    # def get_absolute_url(self):
+    #    return reverse("page_detail", kwargs={"slug": self.slug})
+
+    class Meta:
+        verbose_name = "Detail Page"
+        verbose_name_plural = "Pages"
```
Comment (tahirawan4) [inline models.py]: We won't need this.

File: models.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+        if self.attachment.name.split('.')[-1].lower() \
+                in ['jpg', 'jpeg', 'gif', 'png']:
+            return ('<img height="40" width="60" src="%s"/>' % self.attachment.url)
+        return ('<img height="40" width="60" src="/static/assets/icons/file-icon.png"/>')
+    check_if_image.short_description = 'Attachment'
+    check_if_image.allow_tags = True
+
+    class Meta:
+        verbose_name = 'Detail Gallery'
+        verbose_name_plural = 'Galleries'
```
Comment (tahirawan4) [inline models.py]: This is good.

File: models.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+        ordering = ['-created']
+
+
+class Visitor(TimeStampedModel):
+    post = models.ForeignKey(Post, related_name='post_visitor')
+    ip = models.CharField(max_length=40)
+    isRemove = models.BooleanField(default=False)
```
Comment (tahirawan4) [inline models.py]: Update it.

File: models.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+    def __str__(self):
+        return self.post.title
+
+    class Meta:
+        verbose_name = 'Detail Visitor'
+        verbose_name_plural = 'Visitors'
```
Comment (tahirawan4) [inline models.py]: Remove it.

File: parser/constants.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
+
+
+
```
Comment (tahirawan4) [inline parser/constants.py]: Update all constants with all capitals.

File: parser/models.py [hunk 13]
```diff
@@ -0,0 +1,151 @@
+from django.db import models
```
Comment (tahirawan4) [inline parser/models.py]: Why we have two models files here?
Please remove one and update all the migrations.

File: parser/views.py [hunk 14]
```diff
@@ -2,7 +2,48 @@
...
+    # url(r'^dashboard$', dash_board, name='dash_board'),
+
+    feed_response_data = requests.get(feedly_api_url)
+
+    data = feed_response_data.text
+    return JsonResponse(data, safe=False)
+
+
+def get_categories(request):
+    # TODO: This view is call from the url and will return the Feedly fetch records to html view
```
Comment (tahirawan4) [inline parser/views.py:39]: Why a lot of line breaks here??

**Review by tahirawan4 — CHANGES_REQUESTED** [review]
Please remove all `.`  files committed in the repo.

**Comment by Atizaz-Abid** [comment]
ok
thanks for the review
i will update ,but let me know these model are enough for the v1

**Comment by tahirawan4** [comment]
@aitzazabid @AhmedKakar 
Please resolve conflicts by merging master into this branch,
How to do that.
1) Change your branch to master
2) Pull master
3) Switch Back to this branch
4) And now merge master in this branch

Also models and not good for V1, I need to update the, too. I am exploring feedly. As soon as i got something i will update it.

**Comment by tahirawan4** [comment]
Where are the migrations??? Did you try to run this code?? @aitzazabid @AhmedKakar

File: parser/models.py [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+
+class Feeds(TimeStampedModel):
+    author = models.ForeignKey(Author, related_name='author_post')
+    title = models.CharField(max_length=200)
+    slug = models.SlugField(max_length=200, unique=True)
+    cover = models.ImageField(upload_to='gallery/covers/%Y/%m/%d',
+                              null=True,
+                              blank=True,
+                              help_text='Optional cover post')
+    description = RedactorField()
```
Comment (tahirawan4) [inline parser/models.py]: @aitzazabid  @AhmedKakar  What is this field? How do you import this field??

File: parser/models.py [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+        verbose_name = 'Detail Post'
+        verbose_name_plural = 'Posts'
+        ordering = ["-created"]
+
+
+class Page(TimeStampedModel):
+    author = models.ForeignKey(Author, related_name='author_page')
+    title = models.CharField(max_length=200)
+    slug = models.SlugField(max_length=200, unique=True)
+    description = RedactorField()
```
Comment (tahirawan4) [inline parser/models.py]: Same here?

**Comment by tahirawan4** [comment]
@aitzazabid @AhmedKakar As you are using requests in views, When ever you use a third party module, please add it into requirements file.

**Comment by Atizaz-Abid** [comment]
ok
we didn't run the migration yet ,waiting for the best and approved models

**Review by tahirawan4 — APPROVED** [review]

**Pull request merged**